hazelcast/src/main/java/com/hazelcast/config/JavaKeyStoreSecureStoreConfig.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * Java KeyStore Secure Store configuration.
+ */
+public class JavaKeyStoreSecureStoreConfig
+        extends SecureStoreConfig {
+    /**
+     * The default Java KeyStore type (JCEKS).
+     */
+    public static final String DEFAULT_KEYSTORE_TYPE = "JCEKS";
```

Comment:
We should use the `PKCS12` instead. It's the default type from Java 9 onwards.
Even the `keytool` in newer Java versions suggest switching to `PKCS12` when you are creating `JKS` or `JCEKS` keystores:

```
Warning:
The JCEKS keystore uses a proprietary format. It is recommended to migrate to PKCS12 which is an industry standard format using "keytool -importkeystore -srckeystore vault.keystore -destkeystore vault.keystore -deststoretype pkcs12".
```

I also remember there are some problems with transferring JCEKS from one Java to another (IBM vs Oracle java).

The default keystore type up to Java 8 was `JKS`, but this type doesn't support secret keys. It's able to only handle certificates and keypairs.

Reply:
Thanks for the input. I will change the default to `PKCS12`.

Reply:
Default changed to `PKCS12`